parser/prism/Translator.cc

```diff
@@ -716,6 +749,13 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
             supportedCallType &= supportedBlock;
 
             if (!supportedCallType) {
+                if (prismBlock && PM_NODE_TYPE_P(prismBlock, PM_BLOCK_ARGUMENT_NODE)) {
+                    // PM_BLOCK_ARGUMENT_NODE models the `&b` in `a.map(&b)`,
+                    // but not a literal block with `{ ... }` or `do ... end`
+
+                    args.emplace_back(move(blockPassNode));
+                }
```

Comment:
This chunk is repeated for now, but will naturally disappear when we do the big de-whitequarkifying.